### DISCUSSION TITLE
Extension to integrate lock/unlock functionality directly into the firmware?

### DISCUSSION THREAD

**Discussion by the-smart-home-maker** [rednblkx/HomeKey-ESP32#96] [Ideas]
Hi there,

I would love to be able to directly integrate a lock into the software, so when the key is identified, a lock sitting on a certain GPIO will be unlocked/locked without the need for a separate device. And the lock itself also provides a possibility to read out the lock state. 

I read that release 0.3 includes the "GPIO actions" but I didn't find a documentation for it. Is my use case mentioned above possible?

Best,
The Smart Home Maker

**Comment by rednblkx** [comment]
Hi!

I'm working on improving the wiki, for "GPIO Actions" there is not much to talk about, i only summarized it in the [Web Interface](https://github.com/rednblkx/HomeKey-ESP32/wiki/WebUI) section, basically you can assign one pin that will be pulled LOW/HIGH when you interact in the Home App or on HomeKey. On reboot the pin will be actioned according to the last state.

Also, working on adding a rebound feature for gpio actions that will allow you to make the lock have kind of a momentary state.

**Reply by the-smart-home-maker** [reply]
Hi! 

Thank you so much for the response! Found it in the web interface section now :) 

Such a great project!

Thanks and best regards
The Smart Home Maker

**Reply by the-smart-home-maker** [reply]
I now found the ESPHome integration for this project and that looks even more promising! Awesome work :-)
